mailbox/event/event-memory/src/test/java/org/apache/james/mailbox/events/MemoryEventDeadLettersHealthCheckTest.java

```diff
@@ -0,0 +1,63 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+public class MemoryEventDeadLettersHealthCheckTest implements EventDeadLettersHealthCheckContract {
+    private EventDeadLettersHealthCheck testee;
+    private MemoryEventDeadLetters eventDeadLetters;
+
+    @BeforeEach
+    void setup() {
+        eventDeadLetters = new MemoryEventDeadLetters();
+        this.testee = new EventDeadLettersHealthCheck(eventDeadLetters);
+    }
+
+    @Override
+    public EventDeadLettersHealthCheck testee() {
+        return testee;
+    }
+
+    @Override
+    public EventDeadLetters eventDeadLetters() {
+        return eventDeadLetters;
+    }
+
+    @Override
+    public void createErrorWhenDoingHealthCheck() {
+        // Unable to trigger the EventDeadLetter error with memory version
```

Comment:
IMO calling this method should throw as we are not able to perform the operation. Let me suggest a `NotImplementedException`

Reply:
If that we need to declared at main method at `contract interface` then we can throw at `memory version` but for `Cassandra version` it is useless throws.
Mean while, this method would `never be called` because the test case - which used it were `disabled`.

Reply:
This would allow making sure we are not calling createErrorWhenDoingHealthCheck in the test while this is not implemented. (even if we don't play the tests)

Of course cassandra version should not throw but properly pause/unpause docker.